core/src/main/java/jenkins/model/Nodes.java

```diff
@@ -112,10 +113,12 @@ public void setNodes(final @NonNull Collection<? extends Node> nodes) throws IOE
             @Override
             public void run() {
                 Set<String> toRemove = new HashSet<>(Nodes.this.nodes.keySet());
+                Set<LabelAtom> labels = new HashSet<>();
```

Comment:
this variable appears to not be used (it is write only and disgaurded?

Reply:
Removed. I had initially set up a selective trimLabels here, however since this is setting the whole collection of nodes I figured out this needed a full trim